website/docs/en/api/javascript-api/compilation.mdx

```diff
@@ -410,6 +411,67 @@ compiler.hooks.make.tap('MyPlugin', compilation => {
   </CollapsePanel>
 </Collapse>
 
+### addRuntimeModule
+
+<ApiMeta addedVersion="1.0.6" />
+
+Add a custom runtime module to the compilation.
+
+```ts
+addRuntimeModule(
+  c: Chunk, // the runtime chunk which to add the runtime module
+  m: RuntimeModule, // the runtime module instance to add
+): void;
+```
+
+The following code will add a runtime module which define `__webpack_require__.mock` to the `"main"` chunk:
+
+```js title="rspack.config.js"
+module.exports = {
+  entry: './index.js',
+  plugins: [
+    {
+      apply(compiler) {
+        const { RuntimeModule } = compiler.webpack;
+
+        class CustomRuntimeModule extends RuntimeModule {
+          constructor() {
+            super('custom');
+          }
+
+          generate(compilation) {
```

Comment:
just wondering, is this compilation param a typo?

Webpack types do not have it:
https://github.com/webpack/webpack/blob/main/types.d.ts#L12839

Reply:
Yes, you are right. Pass the compilation as param of `RuntimeModule.generate` is not aligned with webpack. I will fix it in https://github.com/web-infra-dev/rspack/pull/8000